File: src/test/java/com/flexpoker/test/util/CommonAssertions.java
package com.flexpoker.test.util;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;

import java.util.List;
import java.util.UUID;

import com.flexpoker.framework.event.Event;
import com.flexpoker.table.command.aggregate.Table;

public class CommonAssertions {

    public static void verifyEventIds(UUID tableId, List<? extends Event> events) {
        var version = 0;
        for (Event event : events) {
            assertEquals(tableId, event.getAggregateId());
        }
    }

    public static void verifyNumberOfEventsAndEntireOrderByType(
            List<? extends Event> events, Class<? extends Event>... eventClasses) {
        assertEquals(eventClasses.length, events.size());
        assertArrayEquals(eventClasses, events.stream().map(x -> x.getClass()).toArray());
    }

    public static void verifyAppliedAndNewEventsForAggregate(Table table, Class<? extends Event>... eventClasses) {
        verifyEventIds(table.getAggregateId(), table.fetchAppliedEvents());
        verifyEventIds(table.getAggregateId(), table.fetchNewEvents());
        verifyNumberOfEventsAndEntireOrderByType(table.fetchAppliedEvents(), eventClasses);
        verifyNumberOfEventsAndEntireOrderByType(table.fetchNewEvents(), eventClasses);
    }

}

File: src/test/kotlin/com/flexpoker/test/util/CommonAssertions.kt
package com.flexpoker.test.util

import com.flexpoker.framework.event.Event
import com.flexpoker.table.command.aggregate.TableState
import com.flexpoker.table.command.events.TableEvent
import org.junit.jupiter.api.Assertions.assertArrayEquals
import org.junit.jupiter.api.Assertions.assertEquals
import java.util.UUID

object CommonAssertions {

    fun verifyEventIds(tableId: UUID, events: List<Event>) {
        for (event in events) {
            assertEquals(tableId, event.aggregateId)
        }
    }

    fun verifyNumberOfEventsAndEntireOrderByType(events: List<Event>, vararg eventClasses: Class<out Event>) {
        assertEquals(eventClasses.size, events.size)
        assertArrayEquals(eventClasses, events.map { it.javaClass }.toTypedArray())
    }

    fun verifyAppliedAndNewEventsForAggregate(state: TableState, events: List<Event>, vararg eventClasses: Class<out Event>) {
        verifyEventIds(state.aggregateId, events)
        verifyNumberOfEventsAndEntireOrderByType(events, *eventClasses)
    }

    fun verifyNewEvents(aggregateId: UUID, events: List<TableEvent>, vararg eventClasses: Class<out Event>) {
        verifyEventIds(aggregateId, events)
        verifyNumberOfEventsAndEntireOrderByType(events, *eventClasses)
    }

}